<a href="https://colab.research.google.com/github/emerson1000/notebooks/blob/main/M_M_1_M_M_C.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# **MM1**

El sistema M/M/1 es un modelo comúnmente utilizado en la teoría de colas para analizar sistemas de servicio donde los clientes llegan a una tasa exponencial (
λ) y son atendidos a una tasa exponencial (μ) por un solo servidor. El "M" en M/M/1 indica que tanto la llegada de clientes como el servicio se modelan como procesos de Poisson (distribución exponencial).

**Ejemplo:**

Supongamos que gestionamos una línea de ensamblaje en una fábrica de productos electrónicos. Los trabajadores ensamblan componentes en productos terminados, y queremos entender cómo se comporta la producción en esta línea de ensamblaje, especialmente en términos de la tasa de producción y el tiempo promedio de ensamblaje.

Hemos simulado una línea de ensamblaje donde los productos llegan a una tasa exponencial de 0.2 productos por minuto (λ) y son ensamblados a una tasa exponencial de 0.25 productos por minuto (μ).


Vamos a simular la producción en una línea de ensamblaje y calcular medidas de rendimiento, como la tasa de producción y el tiempo promedio de ensamblaje.

Los parámetros:

* Tasa de Llegada de Productos (λ): Representa la velocidad a la que los productos llegan a la línea de ensamblaje desde una fuente externa. En el contexto industrial, esto podría reflejar la tasa a la que los componentes llegan a la línea de ensamblaje desde la cadena de suministro.

* Tasa de Servicio (μ): Indica la velocidad a la que los productos son ensamblados y completados en la línea de ensamblaje. Esto representa la eficiencia de la línea de producción y cuántos productos se pueden ensamblar por unidad de tiempo.

* Tiempo de Simulación: La duración total de la simulación (480 minutos o 8 horas) refleja el período de tiempo durante el cual se quiere evaluar el rendimiento de la línea de ensamblaje.

* Número de Productos Ensamblados: Es una medida de la producción total de productos electrónicos en la línea de ensamblaje durante el tiempo de simulación.

* Tiempo Promedio de Ensamblaje: Representa el tiempo promedio que un producto pasa en el proceso de ensamblaje. Este tiempo es importante para evaluar la eficiencia y la calidad del proceso de producción.

* Tasa de Producción: Indica la cantidad de productos ensamblados por unidad de tiempo y mide la eficiencia de la línea de producción.

Estos parámetros se ajustaron al contexto industrial de la línea de ensamblaje de productos electrónicos para comprender y evaluar la eficiencia y el rendimiento de la producción en esta área específica.

In [ ]:
!pip install simpy

In [ ]:
import simpy
import random

# Configuración de la simulación
env = simpy.Environment()
tiempo_total_simulacion = 480  # Duración total de la simulación en minutos (8 horas)
productos_ensamblados = 0
tiempo_total_ensamblaje = 0

# Parámetros del sistema
tasa_llegada = 0.2  # Tasa de llegada de productos por minuto (clientes/minuto)
tasa_servicio = 0.25  # Tasa de servicio por minuto (productos/minuto)

# Función para el ensamblaje de productos
def ensamblar_producto(env):
    global productos_ensamblados, tiempo_total_ensamblaje
    while True:
        yield env.timeout(random.expovariate(tasa_llegada))  # Tiempo hasta la próxima llegada
        productos_ensamblados += 1
        tiempo_servicio = random.expovariate(tasa_servicio)  # Tiempo de servicio
        tiempo_total_ensamblaje += tiempo_servicio
        yield env.timeout(tiempo_servicio)

# Inicialización del proceso de ensamblaje
env.process(ensamblar_producto(env))

# Ejecución de la simulación
env.run(until=tiempo_total_simulacion)

# Cálculo de medidas de rendimiento
tasa_produccion = productos_ensamblados / tiempo_total_simulacion  # Tasa de producción (productos/minuto)
tiempo_promedio_ensamblaje = tiempo_total_ensamblaje / productos_ensamblados  # Tiempo promedio de ensamblaje

# Resultados
print(f'Tasa de producción: {tasa_produccion:.2f} productos/minuto')
print(f'Tiempo promedio de ensamblaje: {tiempo_promedio_ensamblaje:.2f} minutos/producto')


Tasa de producción: 0.11 productos/minuto
Tiempo promedio de ensamblaje: 3.99 minutos/producto


**Interpretación:**

* Tasa de producción: 0.12 productos/minuto

Esta medida indica la cantidad de productos ensamblados por unidad de tiempo, en este caso, por minuto.
En este ejemplo, la tasa de producción es de 0.12 productos/minuto, lo que significa que, en promedio, se ensamblan 0.12 productos cada minuto.
Esta tasa es un indicador importante de la capacidad de producción del sistema. Una tasa de producción más alta significa que el sistema puede ensamblar más productos en un período de tiempo dado.

* Tiempo promedio de ensamblaje: 3.88 minutos/producto

Esta medida indica cuánto tiempo, en promedio, pasa un producto en el sistema desde que llega hasta que se ensambla completamente.
En este ejemplo, el tiempo promedio de ensamblaje es de 3.88 minutos por producto, lo que significa que, en promedio, un producto pasa aproximadamente 3.88 minutos en el proceso de ensamblaje.
Este valor es relevante para comprender cuánto tiempo se tarda en ensamblar cada producto individualmente. Un tiempo de ensamblaje más corto generalmente indica una mayor eficiencia en el proceso de producción.


En resumen, una tasa de producción de 0.1 productos/minuto sugiere que el sistema está ensamblando productos a un ritmo de 0.1 productos por minuto, y un tiempo promedio de ensamblaje de 3.68 minutos por producto indica cuánto tiempo, en promedio, se necesita para ensamblar un solo producto. Estos valores son críticos para evaluar la eficiencia y el rendimiento de la línea de ensamblaje o el proceso industrial.

# **MMC**

In [ ]:
import simpy
import random

# Configuración de la simulación
env = simpy.Environment()
tiempo_total_simulacion = 480  # Duración total de la simulación en minutos (8 horas)
productos_ensamblados = 0
tiempo_total_ensamblaje = 0
C = 4  # Capacidad de la estación de ensamblaje

# Parámetros del sistema
tasa_llegada = 0.2  # Tasa de llegada de productos por minuto (clientes/minuto)
tasa_servicio = 0.25  # Tasa de servicio por minuto (productos/minuto)

# Función para el ensamblaje de productos
def ensamblar_producto(env, canal):
    global productos_ensamblados, tiempo_total_ensamblaje
    while True:
        yield env.timeout(random.expovariate(tasa_llegada))  # Tiempo hasta la próxima llegada
        productos_ensamblados += 1
        tiempo_servicio = random.expovariate(tasa_servicio)  # Tiempo de servicio
        tiempo_total_ensamblaje += tiempo_servicio
        yield env.timeout(tiempo_servicio)

# Inicialización de los procesos de ensamblaje
for canal in range(C):
    env.process(ensamblar_producto(env, canal))

# Ejecución de la simulación
env.run(until=tiempo_total_simulacion)

# Cálculo de medidas de rendimiento
tasa_produccion = productos_ensamblados / tiempo_total_simulacion  # Tasa de producción (productos/minuto)
tiempo_promedio_ensamblaje = tiempo_total_ensamblaje / productos_ensamblados  # Tiempo promedio de ensamblaje

# Resultados
print(f'Tasa de producción: {tasa_produccion:.2f} productos/minuto')
print(f'Tiempo promedio de ensamblaje: {tiempo_promedio_ensamblaje:.2f} minutos/producto')


Tasa de producción: 0.45 productos/minuto
Tiempo promedio de ensamblaje: 4.10 minutos/producto


**Interpretación de diferencias entre MM1 y MMC:**


* Sistema MM1 (Monocanal) :

Tasa de llegada (λ) = 0.2

Tasa de servicio (μ) = 0.25

Capacidad (C) = 1

Tasa de producción: 0.11 productos/minuto

Tiempo promedio de ensamblaje: 3.99 minutos/producto


* Sistema MMC con C=4:

Tasa de llegada (λ) = 0.2

Tasa de servicio (μ) = 0.25

Capacidad (C) = 4


Tasa de producción: 0.45 productos/minuto

Tiempo promedio de ensamblaje: 4.10 minutos/producto

**Métricas**

* Tasa de producción:

 En el sistema MM1, la tasa de producción es baja (0.11 productos/minuto) debido a la baja tasa de servicio en comparación con la tasa de llegada. Esto significa que los productos llegan más rápido de lo que pueden ser ensamblados, lo que resulta en una cola de espera.

En el sistema MMC con C=4, la tasa de producción es más alta (0.45 productos/minuto) porque hay más canales de servicio disponibles para ensamblar productos.

* Tiempo promedio de ensamblaje:

El tiempo promedio de ensamblaje por producto es un poco más alto en MM4 (4.10 minutos/producto) porque, a pesar de una tasa de producción más alta, los productos pueden experimentar un tiempo de espera más largo antes de ser ensamblados. Sin embargo, una vez que comienza el ensamblaje, es más eficiente debido a la disponibilidad de múltiples canales de servicio.

En resumen, en el sistema MMC con C=4, los productos pueden experimentar menos tiempo de espera antes de ser atendidos gracias a la mayor capacidad de servicio, pero el tiempo promedio de ensamblaje por producto puede ser ligeramente mayor debido a la posible competencia por los canales de servicio disponibles. La elección entre MM1 y MMC con C=4 dependerá de los objetivos y requisitos específicos del sistema y si se valora más la tasa de producción o el tiempo promedio de ensamblaje.







**Ejemplo**

Podemos considerar una situación en la que una fábrica de productos electrónicos tiene una estación de trabajo de prueba de calidad para evaluar la funcionalidad de los productos antes de ser empaquetados y enviados al mercado. La estación de prueba de calidad puede manejar un número limitado de productos a la vez.

Problema: La fábrica desea evaluar la eficiencia de su estación de prueba de calidad y determinar la tasa de producción y el tiempo promedio que un producto pasa en el sistema de prueba de calidad.

Parámetros:

* Tasa de llegada de productos (λ): 20 productos por hora.
* Tasa de servicio (μ): 10 productos por hora.
* Capacidad de la estación de prueba de calidad (C): 2 productos a la vez.

In [ ]:
import numpy as np

# Parámetros del sistema
lambda_ = 20  # Tasa de llegada de productos (productos por hora)
mu = 10       # Tasa de servicio (productos por hora)
C = 2         # Capacidad de la estación de prueba de calidad (número de canales)

# Calcular la tasa efectiva de servicio
rho = lambda_ / (C * mu)

if rho == 1:
    Lq = C / 2
else:
    Lq = (rho ** C) / (np.math.factorial(C) * (1 - rho)) * (rho / (1 - rho))

# Calcular la tasa de producción (productos por hora)
L = lambda_ * Lq

# Calcular el tiempo promedio que un producto pasa en el sistema (en horas)
if rho == 1:
    Wq = 1 / (2 * mu)
else:
    Wq = Lq / lambda_

W = Wq + (1 / mu)

# Mostrar los resultados
print(f"Tasa de producción: {L:.2f} productos por hora")
print(f"Tiempo promedio en el sistema: {W:.2f} horas por producto")

Tasa de producción: 20.00 productos por hora
Tiempo promedio en el sistema: 0.15 horas por producto


# **Sistema de serie**

**Ejemplo:**

Consideremos una empresa de fabricación que tiene una línea de producción de automóviles. La línea de producción está formada por tres estaciones de trabajo:

* Estación 1: Pintura
* Estación 2: Montaje
* Estación 3: Prueba

Los automóviles llegan a la línea de producción a una tasa media de 10 automóviles por hora. Cada estación de trabajo tarda una hora en completar su tarea.

Modelo de colas

Este sistema puede ser modelado como un sistema de colas M/M/C, donde C es el número de estaciones de trabajo en la línea de producción. En este caso, C = 3.

Análisis

Podemos utilizar el algoritmo de Python para calcular las medidas de rendimiento del sistema.

In [ ]:
!pip install simpy

In [ ]:
import simpy
import random

# Parámetros del sistema
tasa_llegada = 10 / 60  # Tasa de llegada de automóviles (automóviles por minuto)
tiempo_procesamiento = 60  # Tiempo de procesamiento en cada estación (en minutos)
numero_estaciones = 3  # Número de estaciones de trabajo en la línea de producción

# Configuración de la simulación
env = simpy.Environment()
cola = simpy.Store(env)  # Corrección: Usar simpy.Store en lugar de simpy.Queue
tiempo_total_produccion = 0
tiempo_total_sistema = 0
num_automoviles_producidos = 0

# Función para el proceso de ensamblaje en cada estación de trabajo
def estacion_trabajo(env, nombre):
    while True:
        llegada_automovil = random.expovariate(tasa_llegada)
        yield env.timeout(llegada_automovil)

        llegada_cola = env.now
        with cola.put(1) as req:  # Usar cola.put(1) para agregar un cliente a la cola
            yield req
            tiempo_proceso = tiempo_procesamiento
            yield env.timeout(tiempo_proceso)

        tiempo_sistema = env.now - llegada_cola
        global tiempo_total_produccion, tiempo_total_sistema, num_automoviles_producidos
        tiempo_total_produccion += tiempo_proceso
        tiempo_total_sistema += tiempo_sistema
        num_automoviles_producidos += 1

# Inicialización de estaciones de trabajo
for i in range(numero_estaciones):
    env.process(estacion_trabajo(env, f'Estación {i + 1}'))

# Ejecución de la simulación
env.run(until=480)  # Duración de la simulación en minutos (8 horas)

# Cálculo de medidas de rendimiento
tasa_produccion = num_automoviles_producidos / 480  # Tasa de producción (automóviles por minuto)
tiempo_promedio_sistema = tiempo_total_sistema / num_automoviles_producidos  # Tiempo promedio en el sistema (minutos)

# Resultados
print(f'Tasa de producción: {tasa_produccion:.2f} automóviles por minuto')
print(f'Tiempo promedio en el sistema: {tiempo_promedio_sistema:.2f} minutos por automóvil')

Tasa de producción: 0.04 automóviles por minuto
Tiempo promedio en el sistema: 60.00 minutos por automóvil


**Interpretación:**

* Tasa de producción: 0.04 automóviles por minuto

Esta medida indica que en promedio se ensamblan 0.04 automóviles por minuto en la línea de producción de automóviles.
Una tasa de producción de 0.04 automóviles por minuto es relativamente baja y sugiere que la línea de producción está ensamblando automóviles a un ritmo lento.

* Tiempo promedio en el sistema: 60.00 minutos por automóvil

Esta medida representa el tiempo promedio que un automóvil pasa en todo el sistema de producción, desde su llegada hasta su ensamblaje completo.
Un tiempo promedio en el sistema de 60.00 minutos por automóvil indica que, en promedio, un automóvil pasa una hora completa en el proceso de producción antes de estar listo.

Un tiempo tan largo en el sistema sugiere que hay un cuello de botella o una ineficiencia en la línea de producción que causa largos tiempos de espera para los automóviles.

La baja tasa de producción y el largo tiempo promedio en el sistema sugieren que la línea de producción de automóviles está experimentando dificultades en términos de eficiencia y capacidad. Esto puede deberse a diversas razones, como problemas de flujo de trabajo, falta de recursos, o tiempos de procesamiento excesivamente largos en una o más estaciones de trabajo.

Para mejorar la eficiencia y aumentar la producción, podría ser necesario identificar y abordar los cuellos de botella en la línea de producción, optimizar los tiempos de procesamiento en las estaciones de trabajo o considerar la adición de recursos adicionales.

En general, estos resultados indican que el proceso de producción actual no está funcionando de manera eficiente y podría requerir ajustes significativos para alcanzar una mayor producción y una reducción en los tiempos de ensamblaje.

# **Sistema de espera Markovianos con fuga de clientes**

**Ejemplo**

Supongamos que estamos gestionando un centro de llamadas y queremos modelar el comportamiento de los clientes que llaman. En este ejemplo, simularemos un sistema de espera Markoviano con fuga de clientes y calcularemos la tasa de abandono esperada.

In [ ]:
import simpy
import random

# Parámetros del sistema
tasa_llegada = 15  # Clientes por hora
tasa_servicio = 20  # Clientes por hora
tasa_abandono = 3  # Clientes por hora
capacidad_sistema = 100  # Capacidad del sistema

# Configuración de la simulación
env = simpy.Environment()
cola = simpy.Store(env)
tiempo_total_abandono = 0
num_clientes_atendidos = 0

# Función para el proceso de atención al cliente con fuga
def servicio_cliente(env):
    global tiempo_total_abandono, num_clientes_atendidos
    while True:
        llegada_cliente = random.expovariate(tasa_llegada)
        yield env.timeout(llegada_cliente)

        tiempo_servicio = random.expovariate(tasa_servicio)
        if len(cola.items) < capacidad_sistema:
            with cola.put(1) as req:
                yield req
                yield env.timeout(tiempo_servicio)
                tiempo_total_abandono += tiempo_servicio
                num_clientes_atendidos += 1
        else:
            tiempo_total_abandono += tiempo_servicio

# Inicialización del sistema
env.process(servicio_cliente(env))

# Ejecución de la simulación
env.run(until=480)  # Duración de la simulación en minutos (8 horas)

# Cálculo de la tasa de abandono esperada
tasa_abandono_esperada = (tasa_abandono / (tasa_llegada + tasa_abandono)) * 100

# Resultados
print(f'Número de clientes atendidos: {num_clientes_atendidos:.2f} clientes')
print(f'Tiempo total servicio: {tiempo_total_abandono:.2f} min')
print(f'Tasa de abandono esperada: {tasa_abandono_esperada:.2f}%')



Número de clientes atendidos: 100.00 clientes
Tiempo total servicio: 358.17 min
Tasa de abandono esperada: 16.67%


**Interpretacion**

Este código ajusta la simulación para simular la fuga de clientes cuando la cola alcanza su capacidad máxima sin usar prioridades. Debería funcionar sin errores y proporcionar resultados precisos para la tasa de abandono esperada.

La salida del algoritmo indica que la "Tasa de abandono esperada" es del 16.67%. Esto significa que, en promedio, el 16.67% de los clientes que llegan al sistema de atención al cliente optan por abandonar la cola en lugar de esperar a ser atendidos. Esta tasa de abandono refleja la probabilidad de que un cliente renuncie debido a la espera o cualquier otro motivo antes de ser atendido.

En el contexto industrial o de atención al cliente, si una tasa de abandono del es alta, podría ser una señal de que el sistema necesita mejoras en términos de capacidad de servicio o eficiencia para reducir la frustración de los clientes y retenerlos en la cola.

La interpretación general es que una tasa de abandono alta puede ser indeseable y requerir acciones correctivas, como la adición de más servidores o la optimización de los tiempos de servicio, para reducir la tasa de abandono y mejorar la satisfacción del cliente.

# **Sistema de esperas con capacidad limitada**

**Ejemplo**

Supongamos que estamos gestionando una línea de producción en una fábrica de componentes electrónicos, y deseamos controlar la cantidad máxima de productos en proceso en cada estación de trabajo. En este ejemplo, simularemos una línea de producción con capacidad limitada en cada estación y calcularemos la eficiencia del sistema.

In [ ]:
import simpy
import random

# Parámetros del sistema
capacidad_estacion = 5  # Capacidad máxima de cada estación
tasa_llegada = 10  # Productos por hora
tasa_servicio = 12  # Productos por hora

# Configuración de la simulación
env = simpy.Environment()
estaciones = [simpy.Resource(env, capacity=capacidad_estacion) for _ in range(3)]
tiempo_total_produccion = 0
num_productos_producidos = 0

# Función para el proceso de producción en cada estación
def produccion_componente(env, estacion_idx):
    global tiempo_total_produccion, num_productos_producidos
    while True:
        llegada_producto = random.expovariate(tasa_llegada)
        yield env.timeout(llegada_producto)

        with estaciones[estacion_idx].request() as req:
            yield req
            tiempo_servicio = random.expovariate(tasa_servicio)
            yield env.timeout(tiempo_servicio)
            tiempo_total_produccion += tiempo_servicio
            num_productos_producidos += 1

# Inicialización de la producción en cada estación
for i in range(3):
    env.process(produccion_componente(env, i))

# Ejecución de la simulación
env.run(until=480)  # Duración de la simulación en minutos (8 horas)

# Cálculo de la eficiencia del sistema
tiempo_total_simulacion = env.now
eficiencia_sistema = (num_productos_producidos / tiempo_total_simulacion) * 60  # Productos por hora

# Resultados
print(f'Eficiencia del sistema: {eficiencia_sistema:.2f} productos por hora')


Eficiencia del sistema: 986.25 productos por hora


**Interpretación:**

Este ejemplo simula una línea de producción en la que cada estación de trabajo tiene una capacidad máxima de 5 productos en proceso. Calcula la eficiencia del sistema en términos de productos producidos por hora. El control de la capacidad en cada estación asegura que el sistema funcione de manera eficiente y evita la congestión.

La salida del algoritmo indica que la "Eficiencia del sistema" es de 975.75 productos por hora. Esta métrica representa la cantidad promedio de productos que se producen en una hora en la línea de producción. Para interpretar este resultado:

Una eficiencia del sistema de 976 productos por hora indica un sistema de producción altamente eficiente y capaz de mantener un flujo constante de producción dentro de las limitaciones de capacidad establecidas.

# **MG1**

* Fábrica de Impresión Digital: Esta fábrica se dedica a la impresión de materiales gráficos y publicitarios.

* Máquina de Impresión: Hay una máquina de impresión digital que se utiliza para imprimir los trabajos.

* Tiempo de Impresión: Cada trabajo de impresión necesita un tiempo diferente para completarse. No todos los trabajos son igualmente complejos ni requieren el mismo tiempo de impresión.

* Llegada de Trabajos: Los trabajos de impresión llegan de manera aleatoria a la máquina de impresión.

In [ ]:
import simpy
import random

# Parámetros del sistema
tiempo_llegada = 10  # Tiempo promedio entre llegadas de trabajos (minutos)
tiempo_servicio_mean = 15  # Tiempo promedio de impresión (minutos)
tiempo_servicio_std = 5  # Desviación estándar del tiempo de impresión (minutos)

# Configuración de la simulación
env = simpy.Environment()
maquina_impresion = simpy.Resource(env, capacity=1)
tiempo_total_impresion = 0
num_trabajos_impresos = 0

# Función para el proceso de impresión
def proceso_impresion(env):
    global tiempo_total_impresion, num_trabajos_impresos
    while True:
        llegada_trabajo = random.expovariate(1.0 / tiempo_llegada)
        yield env.timeout(llegada_trabajo)

        with maquina_impresion.request() as req:
            yield req
            tiempo_servicio = random.normalvariate(tiempo_servicio_mean, tiempo_servicio_std)
            yield env.timeout(tiempo_servicio)
            tiempo_total_impresion += tiempo_servicio
            num_trabajos_impresos += 1

# Inicialización del proceso de impresión
env.process(proceso_impresion(env))

# Ejecución de la simulación
env.run(until=480)  # Duración de la simulación en minutos (8 horas)

# Cálculo de la eficiencia de la máquina de impresión
tiempo_total_simulacion = env.now
eficiencia_maquina = (num_trabajos_impresos / tiempo_total_simulacion) * 60  # Trabajos impresos por hora

# Resultados
print(f'Eficiencia de la máquina de impresión: {eficiencia_maquina:.2f} trabajos por hora')


Eficiencia de la máquina de impresión: 2.38 trabajos por hora


**Interpretación:**

La salida del código proporcionará la eficiencia de la máquina de impresión, que representa la cantidad promedio de trabajos impresos por hora en esta máquina. Esta eficiencia se calcula teniendo en cuenta las llegadas aleatorias de trabajos y los tiempos de impresión variables.

Este ejemplo demuestra cómo utilizar el modelo M/G/1 para simular el proceso de impresión en una fábrica donde los tiempos de impresión no siguen una distribución exponencial típica, sino una distribución general. La eficiencia calculada proporciona información sobre el rendimiento de esta máquina de impresión en condiciones realistas de producción.

La salida del algoritmo indica que la "Eficiencia de la máquina de impresión" es de 2.38 trabajos por hora. La eficiencia se calcula teniendo en cuenta las llegadas aleatorias de trabajos y los tiempos de impresión variables. Si la eficiencia es alta, significa que la máquina está operando a una velocidad considerable y está imprimiendo trabajos con eficiencia.

Por otro lado, una eficiencia más baja podría indicar tiempos de inactividad en la máquina o una menor velocidad de impresión en comparación con la demanda de trabajo.

En resumen, una eficiencia de la máquina de impresión de 2.38 trabajos por hora sugiere que la máquina está funcionando a una velocidad razonable y puede procesar un número significativo de trabajos en una hora. Esta métrica es útil para evaluar el rendimiento de la máquina en un contexto de producción de impresión digital.

# **Fórmula de Pollaczek-Khintchine**

Para ejemplificar su uso en un entorno industrial, imaginemos una fábrica que produce productos electrónicos. Los productos se ensamblan en una línea de producción, y los trabajadores ensamblan los productos individualmente. La tasa de llegada de productos para ensamblar (λ) es de 10 productos por hora, y la tasa de ensamblaje (μ) es de 12 productos por hora. Sin embargo, los tiempos de ensamblaje pueden variar y siguen una distribución normal con una desviación estándar (σ) de 0.5 horas.

Para calcular el tiempo de espera promedio (W) en esta línea de producción utilizando la fórmula de Pollaczek-Khintchine en Python, podemos hacer lo siguiente:

In [ ]:
def calcular_tiempo_espera(lambda_val, mu_val, sigma_val):
    # Calcula el tiempo de espera promedio
    W = 1 / (mu_val - lambda_val) + (lambda_val / (mu_val * (mu_val - lambda_val))) * (sigma_val ** 2)
    return W

# Valores de entrada
lambda_val = 10  # Tasa de llegada de productos por hora
mu_val = 12     # Tasa de ensamblaje de productos por hora
sigma_val = 0.5 # Desviación estándar de los tiempos de ensamblaje en horas

# Calcula el tiempo de espera promedio
tiempo_espera = calcular_tiempo_espera(lambda_val, mu_val, sigma_val)

# Imprime el resultado
print(f"El tiempo de espera promedio en la línea de producción es de {tiempo_espera:.2f} horas.")


El tiempo de espera promedio en la línea de producción es de 0.60 horas.


Esto significa que, en promedio, un producto pasa aproximadamente 0.60 horas (o 36 minutos) esperando en la línea de producción antes de ser ensamblado debido a las variaciones en los tiempos de ensamblaje. Esto podría deberse a factores como la variabilidad en la complejidad de los productos que se ensamblan o a interrupciones inesperadas en la línea de producción.

La información sobre el tiempo de espera promedio es esencial para la gestión de la producción, ya que permite a la empresa optimizar sus procesos, asignar recursos de manera eficiente y garantizar que los productos se ensamblen de manera oportuna para satisfacer la demanda del mercado y mejorar la eficiencia en la línea de producción. En este caso, conocer que el tiempo de espera promedio es de 0.60 horas podría llevar a tomar medidas para reducir este tiempo y mejorar la productividad.